src/Components/Context/TaskProvider.tsx

```diff
@@ -0,0 +1,42 @@
+import { createContext, useState, useEffect, useContext } from "react";
+import type { Task } from "../Types/type";
+
+interface TaskContextProps {
+  tasks: Task[];
+  setTasks: React.Dispatch<React.SetStateAction<Task[]>>;
+}
+
+export const TaskContext = createContext<TaskContextProps | undefined>(
+  undefined
+);
+
+export const useTasks = () => {
+  const context = useContext(TaskContext);
+  if (!context) {
+    throw new Error("useTasks must be used within a TaskProvider");
+  }
+  return context;
+};
+
+const saveTasksToLocalStorage = (tasks: Task[]) => {
+  localStorage.setItem("tasks", JSON.stringify(tasks));
+};
+
+export const TaskProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [tasks, setTasks] = useState<Task[]>(() => {
+    const stored = localStorage.getItem("tasks");
+    return stored ? JSON.parse(stored) : [];
+  });
```

Comment:
This is not the correct way to assign initial value to a setState, I think it has already been discussed in previous PR reviews.
```suggestion
const stored = JSON.parse(localStorage.getItem("tasks")) || [];
  const [tasks, setTasks] = useState<Task[]>(stored);
```

Reply:
Done
commit 729051a86b0e56bba50a0d869fc8d762528d7b19